src/lib/operators/maintenance/create_table.cpp

```diff
@@ -42,20 +43,33 @@ const std::string CreateTable::description(DescriptionMode description_mode) con
   return stream.str();
 }
 
-std::shared_ptr<const Table> CreateTable::_on_execute() {
+const TableColumnDefinitions CreateTable::column_definitions() const {
+  return input_table_left()->column_definitions();
+}
+
+std::shared_ptr<const Table> CreateTable::_on_execute(std::shared_ptr<TransactionContext> context) {
+  const auto column_definitions = _input_left->get_output()->column_definitions();
+
   // If IF NOT EXISTS is not set and the table already exists, StorageManager throws an exception
   if (!if_not_exists || !StorageManager::get().has_table(table_name)) {
     // TODO(anybody) chunk size and mvcc not yet specifiable
     const auto table = std::make_shared<Table>(column_definitions, TableType::Data, Chunk::DEFAULT_SIZE, UseMvcc::Yes);
     StorageManager::get().add_table(table_name, table);
   }
+
+  if (input_table_left()->row_count() > 0) {
```

Comment:
Why this branch? Does Insert crash on empty input? otherwise I'd just let Insert handle this case.

Reply:
If it does crash, please add a brief comment a la "// Insert cannot handle emtpy input"

Reply:
From a reader's perspective, CREATE TABLE has two cases (column definitions or SELECT as an input). If those two cannot be found in the code because `Insert` magically handles both, this is more confusing than helpful.

Reply:
Worse, this `if` does not properly distinguish the two syntaxes of CREATE TABLE - CREATE TABLE ... FROM could return an empty table as well. In any case, doc'ing is warranted, I guess.

Reply:
Good point. We should just doc that both cases can be handled the same.

Reply:
> From a reader's perspective, CREATE TABLE has two cases

Regarding that, I really think "the reader" should be taught to forget about SQL as soon as he enters the LQP/PQP-realm. It's really not helpful to think in SQL terms once the optimizer comes into play. I think we should mentally decouple the CreateTable operator/node from the CREATE TABLE statement - and then it's just fine that there is one path handling both higher level cases.